### PR TITLE
ROX-24484: Add advanced filters and regex to namespace view

### PR DESCRIPTION
**Pull request by dvail**
## Description

Updates the namespace view to use advanced filters and regex search by default.

Also adds a Cypress test to verify that filters are correctly passed to the overview page and displayed as chips. The move to advanced filters broke this subtly and was not detected.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

image
image
image

### CODE REVIEW

File: ui/apps/platform/cypress/integration/vulnerabilities/workloadCves/WorkloadCves.selectors.js [hunk 1]
```diff
@@ -55,6 +55,7 @@ export const selectors = {
     paginationNext: "button[aria-label='Go to next page']",
     severityIcon: (severity) => `span.pf-v5-c-icon:contains('${severity}')`,
     loadingSpinner: 'svg[role="progressbar"][aria-valuetext="Loading..."]',
+    namespaceViewButton: 'button:contains("Prioritize by namespace view")',
```
Comment (pedrottimark) [inline ui/apps/platform/cypress/integration/vulnerabilities/workloadCves/WorkloadCves.selectors.js:58]: An question, not necessarily a change request, about pro and con of selector when a helper function encapsulates access to the DOM element.

Comment (dvail) [reply ui/apps/platform/cypress/integration/vulnerabilities/workloadCves/WorkloadCves.selectors.js:58]: Fair point, and I don't see any reason we would ever need this selector other than via use in that helper function.

File: ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/NamespaceView/NamespaceViewPage.tsx [hunk 2]
```diff
@@ -138,8 +142,27 @@ function NamespaceViewPage() {
...
+                    : selectedSearchFilter.filter((oldValue) => value !== oldValue),
+        };
+
+        if (action === 'ADD') {
+            // TODO - Add analytics tracking
+        }
+
+        setSearchFilter(newFilter);
+        onFilterChange();
+    }
```
Comment (sachaudh) [inline ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/NamespaceView/NamespaceViewPage.tsx:162]: I recently made a function called `onURLSearch` in the `CompoundSearchFilter/utils` file. I think we can take what you did here and use it there. So we could ideally update this function to work like this:

```
const onSearch = ({ category, value, action }: OnSearchPayload) => {
    if (action === 'ADD') {
        // TODO - Add analytics tracking
    }
    onURLSearch(searchFilter, setSearchFilter, payload);
    onFilterChange();
};
```

**Review by sachaudh — APPROVED** [review]
LGTM 🚢

File: ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/NamespaceView/NamespaceViewPage.tsx (hunk 2 above)
Comment (dvail) [reply ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/NamespaceView/NamespaceViewPage.tsx:162]: Ah nice - I have an implementation like this in the AdvancedFiltersToolbar as well. I can refactor these both out in a follow up.

**Comment by dvail** [comment]
Running into an issue with the e2e tests here due to timing. What is happening is that during a CI slowdown, as soon as the test detects that the query has completed it attempts to get the first row of the table. The problem comes from slower executions, where the first row of the table is still the loading spinner, while in faster executions the first row of the table is the actual data we want.

My initial attempt to fix this was to add a check for the table loading spinner, then a second check that the loading spinner was removed directly after the `interactAndWait` call. This worked on the slower executions, but then failed on the full speed executions due to the opposite problem of above -- the spinner had already been removed by the time the first assertion ran.

Here is the log of the original test at full network speed, note the row child count of `6` cells:
image

Here is the log of the original test at 3G, with a row child count of `1` cell, the loading spinner:
image

Here is a log of the test with both `interactAndWait` and the table spinner assertion at full network speed - failure before we get the row count due to removal of the loading spinner before the assertion:
image

Here is a log of the previous at simulated Fast 3G (Slow 3G is identical), again `6` cells:
image

For this particular test, removing the `interactAndWait` call and relying solely on the state of the table appears to fix the issue at all speeds. This is due to the fact the the network delay will always be enough time for the loading spinner to render, which we can assert from the test followed by an immediate assertion that the spinner does not exist. At the point the spinner does not exist we can be sure the next render is one that contains the table data.

It seems like a combination of network speed and rendering speed can cause a race condition between "the request has finished", "the next React render cycle", and Cypress's async querying of the elements. 

I was able to get this to pass consistently locally at full speed, fast 3G, and slow 3G. I'll push an update and see if CI agrees with the next run.

**Pull request merged**